### PR TITLE
Dependency update 07-26

### PR DESCRIPTION
**Pull request by KCeh**
## Summary

I am bumping dependencies and making related changes to keep project up to date

## Changes

- Dependency bump
- Compile and targetSdk bump
- JVM toolchain bump

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [x] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

I removed all Gradle/AGP warning/deprecations I could, there still some left but we are blocked by some plugins etc

### CODE REVIEW

File: config.gradle.kts [hunk 1]
```diff
@@ -6,9 +6,9 @@ allprojects {
 
     extra["buildConfig"] = mapOf(
         "minSdk" to 24,
-        "compileSdk" to 36,
-        "targetSdk" to 36,
-        "buildTools" to "36.0.0"
+        "compileSdk" to 37,
+        "targetSdk" to 37,
+        "buildTools" to "37.0.0"
     )
```
[the remark this answers is not in the document]
Comment (KCeh) [reply config.gradle.kts:12]: Something to keep in mind for the future, not issue right now

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
 junit = "4.13.2"
-kotlin = "2.2.21"
-ksp = "2.3.2"
+kotlin = "2.4.0"
+ksp = "2.3.9"
 leakcanary = "2.14"
-lifecycle = "2.10.0"
-okhttp = "5.3.2"
+lifecycle = "2.11.0"
+okhttp = "5.4.0"
```
[the remark this answers is not in the document]
Comment (KCeh) [reply gradle/libs.versions.toml:33]: This is the latest KSP version, and everything works

**Review by AsimRibo — APPROVED** [review]
Looks good and works. We have some slight issues with edge to edge on demo app and deprecated warnings in console, but we can solve them in separate PR later.
